truss/templates/base.Dockerfile.jinja

```diff
@@ -13,18 +16,19 @@ RUN $PYTHON_EXECUTABLE -c "import sys; \
     || { echo "ERROR: Supplied base image does not have {{ min_supported_python_version_in_custom_base_image }} <= python <= {{ max_supported_python_version_in_custom_base_image }}"; exit 1; }
 {% endblock %}
 
-# NB(nikhil): Only run `pip` if present in image, certain custom servers don't have it.
-RUN if command -v pip >/dev/null 2>&1; then \
-      pip install --upgrade pip --no-cache-dir && rm -rf /root/.cache/pip; \
-    fi
+{% block install_uv %}
+{# Install `uv` if not already present in the image. #}
+RUN if ! command -v uv >/dev/null 2>&1; then (curl -LsSf https://astral.sh/uv/{{ UV_VERSION }}/install.sh | sh) >/dev/null 2>&1; fi
```

Comment:
We'll always need this extra protection, in case the user's base image already has uv installed. However, a future improvement could be to install uv in our own base images, which will save ~2s on a build